xml/admin_ceph_upgrade.xml

```diff
@@ -1397,6 +1397,147 @@ data:
    </step>
   </procedure>
  </sect1>
+
+ <sect1 xml:id="filestore2bluestore">
+  <title>OSD Migration to &bluestore;</title>
+  <para>
+   OSD &bluestore; is a new back end for the OSD daemons. It is the default
+   option since &productname; 5. Compared to &filestore;, which
+   stores objects as files in an XFS file system, &bluestore; can deliver
+   increased performance because it stores objects directly on the underlying
+   block device. &bluestore; also enables other features, such as built-in
+   compression and EC overwrites, that are unavailable with &filestore;.
+  </para>
+  <para>
+   Specifically for &bluestore;, an OSD has a 'wal' (Write Ahead Log) device
+   and a 'db' (RocksDB database) device. The RocksDB database holds the
+   metadata for a &bluestore; OSD. These two devices will reside on the same
+   device as an OSD by default, but either can be placed on different, for
+   example faster, media.
+  </para>
+  <para>
+   In &productname; 5, both &filestore; and &bluestore; are supported and it is possible
+   for &filestore; and &bluestore; OSDs to co-exist in a single cluster. During
+   the &productname; upgrade procedure, &filestore; OSDs are not
+   automatically converted to &bluestore;. Be aware that the
+   &bluestore;-specific features will not be available on OSDs that have not
+   been migrated to &bluestore;.
+  </para>
+  <para>
+   Before converting to &bluestore;, the OSDs need to be running &productname; 5.
+   The conversion is a slow process as all data gets
+   re-written twice. Though the migration process can take a long time to
+   complete, there is no cluster outage and all clients can continue accessing
+   the cluster during this period. However, do expect lower performance for
+   the duration of the migration. This is caused by rebalancing and
+   backfilling of cluster data.
+  </para>
+  <para>
+   Use the following procedure to migrate &filestore; OSDs to &bluestore;:
+  </para>
+  <tip>
+   <title>Turn Off Safety Measures</title>
+   <para>
+    &salt; commands needed for running the migration are blocked by safety
+    measures. In order to turn these precautions off, run the following
+    command:
+   </para>
+ <screen>
+ &prompt.smaster;salt-run disengage.safety
+ </screen>
+   <para>Rebuild the nodes before continuing:</para>
+ <screen>
+ &prompt.smaster; salt-run rebuild.node <replaceable>TARGET</replaceable>
+ </screen>
+   <para>You can also choose to rebuild each node individually. For example:</para>
+ <screen>
+&prompt.smaster; salt-run rebuild.node data1.ceph
+ </screen>
+   <para>The <literal>rebuild.node</literal> always removes and recreates
+     all OSDs on the node.</para>
+     <important>
+       <para>If one OSD fails to convert, re-running the rebuild destroys the
+         already converted &bluestore; OSDs. If this happens, run:</para>
```

Comment:
Is "If this happens" too terse?  Maybe "Instead of re-running the rebuild, you may run:"